### PR TITLE
BP-627 GiftCards: It is possible to paid order by GiftCard more than …

### PR DESCRIPTION
**Pull request by saranchuk-hys**
…order amount value

### CODE REVIEW

File: Controller/Checkout/Giftcard.php [hunk 1]
```diff
@@ -364,7 +365,8 @@ public function execute()
             $this->groupTransaction->saveGroupTransaction($response);
 
             $res['RemainderAmount'] = $response['RequiredAction']['PayRemainderDetails']['RemainderAmount'] ?? null;
-            $alreadyPaid = $this->getAlreadyPaid($orderId) + $response['AmountDebit'];
+            // $alreadyPaid = $this->getAlreadyPaid($orderId) + $response['AmountDebit'];
```
Comment (vladislav-padalka-hys) [inline Controller/Checkout/Giftcard.php:368]: please leave our commented lines 
git has all history if needed

File: Helper/PaymentGroupTransaction.php [hunk 2]
```diff
@@ -116,6 +116,15 @@ public function getGroupTransactionAmount($order_id){
         return $total;
     }
 
+    public function getGroupTransactionOriginalTransactionKey($order_id){
```
Comment (vladislav-padalka-hys) [inline Helper/PaymentGroupTransaction.php:119]: please obey magento code style : space before bracket

File: Helper/PaymentGroupTransaction.php [hunk 3]
```diff
@@ -116,6 +116,15 @@ public function getGroupTransactionAmount($order_id){
...
+        foreach ($this->getGroupTransactionItems($order_id) as $key => $value) {
+            if($value['relatedtransaction']){
```
Comment (vladislav-padalka-hys) [inline Helper/PaymentGroupTransaction.php:121]: please obey magento code style : spaces

File: Model/Push.php [hunk 4]
```diff
@@ -1471,6 +1474,16 @@ private function isGroupTransactionPart()
         return false;
     }
 
+    private function isGroupTransactionFailed()
+    {
+        if ($this->isGroupTransactionInfoType()) {
+            if ($this->postData['brq_statuscode'] == 490) {
```
Comment (florinel-chis) [inline Model/Push.php:1480]: can we move the `490` to a constant?

**Review by vladislav-padalka-hys — APPROVED** [review]

**Pull request merged**